src/github/utils.ts

```diff
@@ -1110,13 +1109,98 @@ export function hasEnterpriseUri(): boolean {
 	return !!getEnterpriseUri();
 }
 
-export function generateGravatarUrl(gravatarId: string | undefined, size: number = 200): string | undefined {
-	return !!gravatarId ? `https://www.gravatar.com/avatar/${gravatarId}?s=${size}&d=retro` : undefined;
+function generateGravatarUrl(gravatarId: string | undefined, size: number = 200): string | undefined {
+	if (!gravatarId) {
+		return undefined;
+	}
+
+	// This URL is a publicly available render of the `resources/icons/dark/github.svg` as a 2048x2048 png
+	const invertocatEncodedUrl = encodeURIComponent('https://user-images.githubusercontent.com/675956/216480776-611f6c3a-6334-46d3-8f5d-9a11ae62df6d.png');
+	const invertocatStyle = 'invertocat';
+
+	let style = vscode.workspace.getConfiguration('githubPullRequests').get<string>('defaultGravatarsStyle', invertocatStyle);
+	if (style === invertocatStyle) {
+		style = invertocatEncodedUrl;
+	} else if (style === 'none') {
+		return undefined;
+	}
+
+	return `https://www.gravatar.com/avatar/${gravatarId}?s=${size}&d=${style}`;
 }
 
-export function getAvatarWithEnterpriseFallback(avatarUrl: string, email: string | undefined, authProviderId: AuthProvider): string | undefined {
-	return authProviderId === AuthProvider.github ? avatarUrl : (email ? generateGravatarUrl(
-		crypto.createHash('md5').update(email?.trim()?.toLowerCase()).digest('hex')) : undefined);
+const enterpriseAvatarQueue = new PQueue({concurrency: 3});
+const enterpriseAvatarCache: {[k: string]: Promise<string | void>} = {};
```

Comment:
This only caches enterprise avatars in memory. This may not be desirable if we want to keep memory usage low. There is an opportunity to improve this static-asset caching by saving these images into the extension's storage URI on the filesystem and then having the webviews serve them from that `vscode-resource:` location. That would avoid having to re-fetch the avatars every time the window reloads.